structuredevent-service-cdp/src/main/java/com/sequenceiq/cloudbreak/structuredevent/repository/CDPStructuredEventRepository.java

```diff
@@ -68,4 +71,12 @@ default List<ResourceBasicView> findAllResourceBasicViewByNamesAndAccountId(Coll
         throw new UnsupportedOperationException();
     }
 
+    @Modifying
+    @Query(value = "DELETE FROM cdp_structured_event cdse WHERE cdse.resourceCrn = :resourceCrn AND cdse.timestamp <= :timestamp", nativeQuery = true)
+    void deleteByResourceCrnOlderThan(@Param("resourceCrn") String resourceCrn, @Param("timestamp") Long timestamp);
+
+    @Modifying
+    @Query(value = "DELETE FROM cdp_structured_event cdse WHERE cdse.resourceCrn LIKE :accountId AND cdse.timestamp <= :timestamp", nativeQuery = true)
```

Comment:
Why do you use nativeQuery here?

Reply:
Also the lie query and accountId param are strange. I think you should add the `%` here in the query, or do you use different search queries here?